src/common/set.c

```diff
@@ -3858,6 +3865,55 @@ util_pool_open_nocheck(struct pool_set *set, unsigned flags)
 	return -1;
 }
 
+/*
+ * util_read_compat_features -- (internal) read compat features from the header
+ */
+static int
+util_read_compat_features(struct pool_set *set, uint32_t *compat_features)
```

Comment:
Do we really need those functions for flags reading? Can't we read them once at open time and cache somewhere?